src/algokit/cli/dispenser.py

```diff
@@ -0,0 +1,214 @@
+import enum
+import logging
+from dataclasses import dataclass
+from pathlib import Path
+
+import click
+
+from algokit.core.dispenser import (
+    DISPENSER_ACCESS_TOKEN_KEY,
+    DispenserApiAudiences,
+    clear_dispenser_credentials,
+    get_oauth_tokens,
+    is_authenticated,
+    process_dispenser_request,
+    revoke_refresh_token,
+    set_dispenser_credentials,
+)
+from algokit.core.utils import is_network_available
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class DispenserAsset:
+    asset_id: int
+    decimals: int
+    description: str
+
+
+class OutputMode(enum.Enum):
+    STDOUT = "stdout"
+    FILE = "file"
+
+
+class DispenserAssetName(enum.IntEnum):
+    ALGO = 0
+
+
+assets = {
+    DispenserAssetName.ALGO: DispenserAsset(
+        asset_id=0,
+        decimals=6,
+        description="Algo",
+    ),
+}
+
+
+def _handle_ci_token(output_mode: str, output_filename: str, token_data: dict) -> None:
+    if output_mode == OutputMode.STDOUT.value:
+        click.echo(f'\n{DISPENSER_ACCESS_TOKEN_KEY} (valid for 30 days):\n\n{token_data["access_token"]}\n')
+        logger.warning(
+            "Your CI access token has been printed to stdout.\n"
+            "Please ensure you keep this token safe!\n"
+            "If needed, clear your terminal history after copying the token!"
+        )
+    else:
+        with Path.open(Path(output_filename), "w") as token_file:
+            token_file.write(token_data["access_token"])
+        logger.warning(
+            f"Your CI access token has been saved to `{output_filename}`.\n"
+            "Please ensure you keep this file safe or remove after copying the token!"
+        )
+
+
+class DispenserGroup(click.Group):
+    def get_command(self, ctx: click.Context, cmd_name: str) -> click.Command | None:
+        return_value = super().get_command(ctx, cmd_name)
+
+        if return_value is None:
+            return None
+        elif is_network_available():
+            return return_value
+        else:
+            logger.error("Please connect to internet first")
+            raise click.exceptions.Exit(code=1)
+
+
+@click.group("dispenser", cls=DispenserGroup)
+def dispenser_group() -> None:
+    """Interact with the AlgoKit TestNet Dispenser."""
+
+
+@dispenser_group.command("logout", help="Logout of your Dispenser API account.")
+def logout_command() -> None:
+    if is_authenticated():
+        try:
+            revoke_refresh_token()
+            clear_dispenser_credentials()
+        except Exception as e:
+            logger.debug(f"Error logging out {e}")
+            raise click.ClickException("Error logging out") from e
+        logger.info("Logout successful")
+    else:
+        logger.warning("Not logged in! To login, run `algokit dispenser login`")
```

Comment:
```suggestion
        logger.info("Already logged out.")
```

Chances are if the user has run logout, they want to log out or confirm they are logged out. I don't really see any value in telling them how to login here.